examples/vm_os_upgrade/05_clone_from_source.yml

```diff
@@ -0,0 +1,32 @@
+- name: Create snapshot for source VM
+  scale_computing.hypercore.vm_snapshot:
+    state: present
+    vm_name: "{{ source_vm_name }}"
+    label: "{{ source_snapshot_label }}"
+  register: vm_created_snapshot
+
+- name: Clone source VM
+  scale_computing.hypercore.vm_clone:
+    vm_name: "{{ cloned_vm_name }}"
+    source_vm_name: "{{ source_vm_name }}"  # latest snapshot will be cloned
+  register: vm_cloned
```

Comment:
Here we should also demonstrate how to select a different VLAN for new VM.
I realize in out testbed diffrent VLAN will not work (there will be no GW for new VLAN).
But we can still provide extra parameters:
- source_vm_vlan: 0
- cloned_vm_vlan: 0

Then it will be easy to see how the VLAN values can be different.

I think you will need to use vm_nic module to change VLAN for the cloned VM NIC.